### ISSUE TITLE
IA 2.6.8 has broken stream that work on IA 2.4.5

### ISSUE THREAD

**Issue by Testato** [peak3d/inputstream.adaptive#615]
Our addon LA7-LA7D (present on Official Kodi Repo) cannot work on last IA for Kodi19.

The original Live channel is here: 
https://www.la7.it/dirette-tv
p.s. it's free, without registration, and not geoblocked

We tested it by old IA on Kody18 and it work, for example on IA 2.4.5 and also on IA 2.3.3

**Comment by rafhtl** [comment]
Same problem here
please help
thankyou

**Comment by glennguy** [comment]
We would require some debug logs - one where it works in Kodi 18 and one where it fails in Kodi 19. I can see `ERROR <general>: AddOnLog: inputstream.adaptive: Download https://d15umi5iaezxgx.cloudfront.net/OUT/DASH/Live_960x540_000066401.mp4 failed with error: 404` from the linked issue but debug logging is off (plus that log snippet doesn't have line endings, horrible to read!) Preferable these logs are captured at very similar times, easier to see the difference in segment calculation.

Also please provide the mpd file from this time as well, thanks!

**Comment by Testato** [comment]
I will provide all the logs today.
Thanks for the interest.

But I have a question: the problem on LA7 addon is not random, and it is present on all platform. So instead of read my logs why not read the log produced on your pc yourself ?

I repeat, i will provide all the log that you need, but my question is only for understand.
If someone open on one of my sw an issue I prefer create all the log myself for use it by my tool, etc.

**Comment by glennguy** [comment]
Fair enough - I am realising now that the content is not geoblocked (most content is)
However you can increase your chances of getting help with issues like these if you do the groundwork and present as much information as possible. Issues that are written as "It's broken, please fix" with little to no information are likely to be ignored by most contributors. 
To put another scenario, if 5 developers look at this issue and all have to recreate the environment (Kodi 18/19, IA versions x/y) and recreate the issue, it's a lot of duplicate work whereas if you were to provide clear logs and documentation/assets it's much easier for someone to quickly jump in for a look. 
Often the underlying cause can be obvious without having to recreate locally. There are plenty of examples in this repository where I (and no doubt others) have been able to assist on issues using supplied logs/manifests where content is geoblocked and the issue unable to be reproduced locally.
In this case I will produce my own logs, later this week I hopefully have time.

**Comment by albert51k** [comment]
Salve,
non sono molto competente in tema KODI.
Da un bel po' non riuscivo più a vedere la diretta LA7 su WIN 10. Avevo **IA 2.4.6**
Leggendo qui... ho provato a recuperare **IA 2.4.5**. (Autore: **peak3d**) e confermo che ora funziona.
La mia versione di KODI è:
**18.6 Git:20200229-8e967df921**
Ho lanciato più o meno tutte le voci presenti nel menu LA7 e, tranne "Programmi on demand" e, qui e là, qualcosa sotto la voce "Telegiornali e meteo" sembra che tutto il resto funzioni (ad es., Dirette La7, LA7d e relativi Rivedi...)

Se utile... questo è quanto :-)
Saluti

**Comment by Testato** [comment]
@albert51k 
I can reassume your message for non italian people: the problem is not the version of Kodi but the version of IA.
On Kodi 18 the issue is present from ver 2.4.6
On Kodi 19 from the beginning, with ver 2.6.7

**Comment by Testato** [comment]
From this morning also the second Live channel, LA7d Live, not work on IA 2.6.7

The addresses received are fixed, it never change.

For "LA7 Live" we receive always Live_640x360_-00000001.mp4
https://www.la7.it/dirette-tv
`ERROR: AddOnLog: InputStream Adaptive: Download https://d15umi5iaezxgx.cloudfront.net/OUT/DASH/Live_640x360_-00000001.mp4 failed with error: 404`

For "LA7d Live" we receive always Live_640x360_000323977.mp4
https://www.la7.it/live-la7d
`ERROR: AddOnLog: InputStream Adaptive: Download https://d15umi5iaezxgx.cloudfront.net/LA7D/OUT/DASH/Live_640x360_000323977.mp4 failed with error: 404`

**Comment by matthuisman** [comment]
try with IA builds from below and see if works:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-564/1/artifacts
and also try these builds:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-564/3/artifacts

Let me know which work / don't work etc

**Comment by Testato** [comment]
I'm on Linux (Ubuntu 20), there is .deb ? Or ppa ?

**Comment by glennguy** [comment]
Just tested Matt and can confirm that the issue is fixed in your PR, specifically by reverting the changes made in DASHTree.cpp in this commit/block [here](https://github.com/xbmc/inputstream.adaptive/commit/dc7b42783d18c65ebcf205a4979a1c9ea6b825e7#diff-f5e439696055d30c282716f0d0fc54ee6612edb6e4f29ef91b4c3a00cd426679L1242-L1256)

Ideally we can get a test case against this issue, but also the original issue that was fixed by the changes - #439
Luckily for us this case is already in the gtest pr. The manifest is `segtimeline_live_ast.mpd` and the test is [here](https://github.com/glennguy/inputstream.adaptive/blob/new-tests/src/test/TestDASHTree.cpp#L120)

**Comment by Testato** [comment]
I tested on Win10 and last IA 2.6.8 and the problem is also present 

Now i test your build @matthuisman

**Comment by Testato** [comment]
> try with IA builds from below and see if works:
> https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-564/1/artifacts

Sorry but I have only Linux and Windows machine

> and also try these builds:
> https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-564/3/artifacts

Tested on Win10
 ~~The audio is fixed~~ The audio do not have 404 error but is not the correct audio
The video continue to do 404 error
```
2021-03-27 13:43:47.116 T:17580    INFO <general>: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder aac
2021-03-27 13:43:47.234 T:3288    ERROR <general>: AddOnLog: inputstream.adaptive: Download https://d15umi5iaezxgx.cloudfront.net/OUT/DASH/Live_640x360_000508221.mp4 failed with error: 404
```

I see that on IA 2.6.8 the received video link is always the same 00000001.mp4
By your build the received video instead is started to becomeis dinamic, so this is good 000508301.mp4

p.s. if you want test this bug using my plugin, you need the last version from my repo, because the ones present on kodi repo is an old version
https://github.com/luivit/plugin.video.rivedila7/archive/refs/heads/master.zip

**Comment by matthuisman** [comment]
@Testato 
Please try:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-564/7/artifacts

That works for me when using the add-on you provided above
**Please confirm**

**Comment by Testato** [comment]
@matthuisman 
YES is work.
Tested on Win10.

Thanks a lot for your effort

Will these fixes be ported to the IA 2.4.6 version ? Or do we have to close support for kodi18 on our plugin ?

**Comment by matthuisman** [comment]
Waiting on other tests before can merge 19.
Dont think 18 getting any more updates...

**Comment by dagwieers** [comment]
That is unfortunate as lots of users cannot update to Matrix (yet).

**Comment by matthuisman** [comment]
I was just assuming...
I'm happy to back port if team willing to do Leia releases.

**Comment by matthuisman** [comment]
@dagwieers 
Also, are you able to test above build with any of your live content to make sure it doesn't break anything? Be great to get it tested as much as  possible.

**Comment by Testato** [comment]
I tested it on all my content, two live channels and some widevine ondemand prigrams, and it work on Win10.

Instead I tested on a certified Widevine L1 Android phone (Samsung S9+) and we have no audio and no video.
But also on IA 2.4.5 the problem is present.

It is a different issue ?

**Comment by matthuisman** [comment]
That's different issue. This only relates to live content and wrong segments

**Comment by fawkes15** [comment]
tested on AndroidTV 9 and Kodi 19, works fine.

Thanks

**Comment by light-wizard** [comment]
Tested version 2.6.8 on Android aarch64. Working fine now. Thanks for the fix.

**Comment by rafhtl** [comment]
I would like to try on RPI4 kodi19
is there a version I can use please?

**Comment by rafhtl** [comment]
I tested with a .deb compiled version and it does work on kodi 19 rpi4 xbian
thank you

**Comment by Testato** [comment]
when the new version will be available on Ubuntu ?

**Comment by matthuisman** [comment]
Kodi 19? I assume it should be already available on the Ubuntu ppa

**Comment by Testato** [comment]
no, it's not,
and also on Android is not present
(i mean for kodi19)

Only on Windows i received the update to 2.6.11

**Comment by matthuisman** [comment]
yes, Android is currently only built-in (not via repo) - its been worked on.
Not sure about the ubunta ppa sorry. it might pop up on the next version as i think it had a issue building

**Comment by phunkyfish** [comment]
@wsnipex can you kick off a manual build of 2.6.11 for the PPA?

**Comment by matthuisman** [comment]
@Testato
Ubuntu ppa should now be updated
And also Android update out

**Comment by Testato** [comment]
it does not seem

```
apt-cache policy kodi-inputstream-adaptive
kodi-inputstream-adaptive:
  Installed: 2.6.8-1~focal
  Candidate: 2.6.8-1~focal
  Version table:
 *** 2.6.8-1~focal 500
        500 http://ppa.launchpad.net/team-xbmc/ppa/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Comment by Testato** [comment]
yep thanks, 2.6.13 

```
apt-cache policy kodi-inputstream-adaptive
kodi-inputstream-adaptive:
  Installed: 2.6.8-1~focal
  Candidate: 2.6.13-1~focal
```

**Comment by matthuisman** [comment]
Saw your previous comment in emails. Did you just forget to apt-get update?

**Comment by Testato** [comment]
yep :-)

also on Kodi for Android is arrived the new version. it is automatically updated at first boot

**Comment by Testato** [comment]
Today the problem has reappeared but now, on IA 2.6.13, the stream of the two live channels start, but only the first segment is played, from the second onwards the segment have error 404.

Do you prefer reopen this issue or a new issue ?

**Comment by matthuisman** [comment]
if 2.6.13 was working and now isn't - then i'd suggest something has changed on their end?

**Comment by Testato** [comment]
Probably, but by browser all.work regularly, so also IA should ?

**Comment by matthuisman** [comment]
For a start - I see it's using SegmentTemplate now... so you'll need to set manifest_update_param to "full" so it refreshes the MPD

```
if live:
    #listitem.setLabel(titolo_diretta)
    listitem.setInfo('video', {'plot': titolo_diretta, 'title': titolo_diretta})
    listitem.setProperty('inputstream.adaptive.manifest_update_parameter', 'full')
```

i actually have a PR to automatically do this I made last night: https://github.com/xbmc/inputstream.adaptive/pull/645

Even with the above - it seems to pause with a lot of "Begin WaitForSegment stream 1" so maybe another issue needs looked into

**Probably best to open a new issue after you have set the above update_parameter.**

**Comment by albert51k** [comment]
Salve,
sempre dall'alto della mia ignoranza in materia, comunico la mia esperienza :-)

Win 10 Home (20H2).
Ho fatto upgrade a Kodi 19.0 (19.0.0) Git:20210218-f44fdfbf67
LA7-LA7d (Luivit, Testato) 5.2.0
InputStream Adaptive (peak3d) 2.6.13
**La diretta parte ...ma si arresta sempre dopo 10-15 secondi.**

Vi seguo con interesse.
Saluti

**Comment by Testato** [comment]
Non scrivere qui in italiano.
Questo serve per IA.
C'è una nuova versione di test sul repo di LA7, provala e scrivi lì eventuali esperienze
https://github.com/luivit/plugin.video.rivedila7/issues/55

**Comment by rafhtl** [comment]
https://github.com/luivit/plugin.video.rivedila7/issues/55

This works on kodi 19
its not a problem with IA
https://github.com/luivit/plugin.video.rivedila7/archive/refs/heads/master.zip
fixes the issue
thank you
